src/Form/Form.php

```diff
@@ -107,7 +118,7 @@ protected function _buildSchema(Schema $schema)
     public function validator(Validator $validator = null)
     {
         if ($validator === null && empty($this->_validator)) {
-            $validator = $this->_buildValidator(new Validator());
+            $validator = $this->_buildValidator(new $this->_validatorClass);
```

Comment:
This has been taken care of in `3.next` by making `Form` use `ValidatorAwareTrait`.